src/profiler.h

```diff
@@ -69,23 +69,11 @@ class Profiler
 	void avg(const std::string &name, float value)
 	{
 		JMutexAutoLock lock(m_mutex);
-		{
-			std::map<std::string, int>::iterator n = m_avgcounts.find(name);
-			if(n == m_avgcounts.end())
-				m_avgcounts[name] = 1;
-			else{
-				/* No add shall have been used */
-				assert(n->second != -2);
-				n->second = MYMAX(n->second, 0) + 1;
-			}
-		}
-		{
-			std::map<std::string, float>::iterator n = m_data.find(name);
-			if(n == m_data.end())
-				m_data[name] = value;
-			else
-				n->second += value;
-		}
+		int& count = m_avgcounts[name];
+		
+		assert(count != -2);
+		count = MYMAX(count, 0) + 1;
+		m_data[name] += value;
```

Comment:
m_avgcounts[name] will insert an element if it doesn't exist and initialise it to the default value, which is T(). This is 0 for number types.